test/generate-testcase.js

```diff
@@ -0,0 +1,117 @@
+var path = require("path");
+var fs = require("fs");
+var jsdom = require("jsdom").jsdom;
+var serializeDocument = require("jsdom").serializeDocument;
+var http = require("http");
+var vm = require("vm");
+var readabilityPath = path.join(__dirname, "..", "Readability.js");
+var jsdomPath = path.join(__dirname, "..", "JSDOMParser.js");
+var ctx = {};
+ctx.dump = console.log
+ctx.console = console;
+vm.runInNewContext(fs.readFileSync(jsdomPath), ctx, jsdomPath);
+vm.runInNewContext(fs.readFileSync(readabilityPath), ctx, readabilityPath);
+var Readability = ctx.Readability;
+var JSDOMParser = ctx.JSDOMParser;
+
+
+if (process.argv.length < 3) {
+  console.log("Need at least a destination slug and potentially a URL (if the slug doesn't have source).");
+  return 0;
+}
+
+var slug = process.argv[2];
+var url = process.argv[3];
+
+var destRoot = path.join(__dirname, "test-pages", slug);
+
+fs.mkdir(destRoot, function(err) {
+  if (err) {
+    var sourceFile = path.join(destRoot, "source.html");
+    fs.exists(sourceFile, function(exists) {
+      if (exists) {
+        fs.readFile(sourceFile, {encoding: "utf-8"}, function(err, data) {
+          if (err) {
+            console.error("Source existed but couldn't be read?");
+            return;
+          }
+          onGotSource(data);
+        });
+      } else {
+        fetchSource();
+      }
+    });
+    return;
+  }
+  fetchSource();
+});
+
+function fetchSource() {
+  if (!url) {
+    console.error("You should pass a URL if the source doesn't exist yet!");
+    return;
+  }
+  var client = http;
```

Comment:
Ah, but a few lines down we use "https" if the URL requires it (because http.get complains if you pass it an https URL and refuses to work, because... reasons, I guess?). So I figured I would make the object we use to do the request use a different name than one specific to the protocol...